docs/cvl/invariants.md

```diff
@@ -205,7 +207,22 @@ we are checking for preservation.
 To see why this is not the desired behavior, consider a `deposit` method that
 increases the message sender's balance.  When the
 `zero_address_has_no_balance_v2` invariant is checked on `deposit`, the Prover
-will report a violation with the `msg.sender` set to 0 in the call to `deposit`
+will effectively check the following (see {ref}`invariant-as-rule`):
```

Comment:
Okay, I see *what* is happening here. I may very well not see something obvious here, but *why* do we have both `e` and `calledEnv`? Maybe there should be some more general explanation (independent of the `preserved` stuff) of how `invariants` behave across multiple messages?

Reply:
I think this is covered in the section on invariant assumptions; I added a link here to that section.